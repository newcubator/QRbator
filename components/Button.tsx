import { Ionicons } from "@expo/vector-icons"; // Assuming Ionicons are used based on home.tsx
import React from "react";
import { Text, TextStyle, TouchableOpacity, ViewStyle } from "react-native";

interface ButtonProps {
  title: string;
  onPress: () => void;
  style?: ViewStyle;
  textStyle?: TextStyle;
  icon?: keyof (typeof Ionicons)["glyphMap"];
  iconColor?: string;
  iconSize?: number;
  className?: string;
  type?: "primary" | "secondary" | "danger";
  disabled?: boolean;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  style,
  textStyle,
  icon,
  iconColor,
  iconSize = 18,
  className,
  type = "primary",
  disabled = false,
}) => {
  const isPrimary = type === "primary";
  const isSecondary = type === "secondary";
  const isDanger = type === "danger";

  let bgColor = "bg-corp-dark-teal";
  let textColor = "text-corp-white";
  let defaultIconColor = "#FFFFFF";
  let border = "";

  if (isSecondary) {
    bgColor = "bg-corp-white";
    textColor = "text-corp-grey";
    defaultIconColor = "#50505E";
    border = "border border-corp-grey";
  } else if (isDanger) {
    bgColor = "bg-corp-dark-red";
    textColor = "text-corp-white";
    defaultIconColor = "#FFFFFF";
  }

  const finalIconColor = iconColor || defaultIconColor;
  const opacity = disabled ? "opacity-50" : "opacity-100";

  return (
    <TouchableOpacity
      style={[style]}
      className={`flex-row items-center justify-center rounded-lg px-4 py-3 ${bgColor} ${border} ${opacity} ${
        className || ""
      }`}
      onPress={onPress}
      activeOpacity={disabled ? 1 : 0.7}
      disabled={disabled}
    >
      {icon && <Ionicons name={icon} size={iconSize} color={finalIconColor} />}
      <Text style={[textStyle]} className={`ml-2 font-semibold ${textColor}`}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};
