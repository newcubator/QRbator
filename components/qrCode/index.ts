export * from "./EmailForm";
export * from "./FormProvider";
export * from "./TagInput";
export * from "./TextForm";
export * from "./TypeSelector";
export * from "./URLForm";
export * from "./VCardForm";
export * from "./WiFiForm";
