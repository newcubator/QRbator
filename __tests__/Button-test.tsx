import { fireEvent, render } from "@testing-library/react-native";
import { Button } from "../components/Button";

describe("<Button />", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with title", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Test Button" onPress={onPressMock} />
    );

    expect(getByText("Test Button")).toBeTruthy();
  });

  test("calls onPress when pressed", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Test Button" onPress={onPressMock} />
    );

    fireEvent.press(getByText("Test Button"));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  test("renders with icon", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Test Button" onPress={onPressMock} icon="star" />
    );

    expect(getByText("Test Button")).toBeTruthy();
  });

  test("applies secondary styling correctly", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Secondary Button" onPress={onPressMock} type="secondary" />
    );

    const button = getByText("Secondary Button");
    expect(button).toBeTruthy();

    expect(button.props.className).toContain("text-corp-grey");
  });

  test("applies danger styling correctly", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Danger Button" onPress={onPressMock} type="danger" />
    );

    const button = getByText("Danger Button");
    expect(button).toBeTruthy();
  });

  test("is disabled when disabled prop is true", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Disabled Button" onPress={onPressMock} disabled={true} />
    );

    const button = getByText("Disabled Button");
    expect(button).toBeTruthy();

    fireEvent.press(button);
    expect(onPressMock).not.toHaveBeenCalled();
  });

  test("matches snapshot", () => {
    const tree = render(
      <Button title="Snapshot Test" onPress={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
