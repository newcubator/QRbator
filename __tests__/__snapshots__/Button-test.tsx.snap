// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    className="ml-2 font-semibold text-corp-white"
    style={
      [
        undefined,
      ]
    }
  >
    Snapshot Test
  </Text>
</View>
`;
